bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -619,9 +622,10 @@ void doCompactEntryLogs(double threshold, long maxTimeMillis) throws EntryLogMet
                                 meta.getEntryLogId(), meta.getUsage(), threshold);
                     }
 
-                    long priorRemainingSize = meta.getRemainingSize();
+                    long compactSize = meta.getTotalSize() - meta.getRemainingSize();
+                    compactionReadByteRateLimiter.acquire((int) (compactSize));
                     compactEntryLog(meta);
```

Comment:
If I understand correctly, when the compact is in progress, if compactionRateBytes is configured, the compact will be limited when writing the disk. Since read and write are processed strictly serially, the next read needs to wait for the write operation to complete. Does it mean that read is also limited to compactionRateByBytes？

I suspect that your problem here is because the disk read rate is too fast caused by getting entryLogMeta from entryLogFile, you can look at this PR: #2963 

@hangc0276 Please help take a look. Thanks.

Reply:
@HQebupt Can you search in your bookie service log to see if there is the following log when the read IO is high: “Extracting entry log meta from entryLogId”

Reply:
Good catch, please see this comment @HQebupt

Reply:
@wenbingshen It did not help after I cherry pick https://github.com/apache/bookkeeper/pull/2963. We can discuss it when you are available tomorrow.
```
// Set the rate at which compaction will readd entries. The unit is bytes added per second.
 compactionRateByBytes=31457280
```
![image](https://github.com/apache/bookkeeper/assets/4970972/f1bbc22b-c260-4fe8-bcf7-a65ab4373243)

![image](https://github.com/apache/bookkeeper/assets/4970972/9ec7df87-bc1b-4137-b4ac-9f7476b99d9a)